app/models/supporter.rb

```diff
@@ -55,6 +65,9 @@ class Supporter < ActiveRecord::Base
   belongs_to :merged_into, class_name: 'Supporter', :foreign_key => 'merged_into'
   has_many :merged_from, class_name: 'Supporter', :foreign_key => "merged_into"
 
+  has_many :addresses, class_name: "SupporterAddress"
+  belongs_to :primary_address, class_name: "SupporterAddress"
```

Comment:
Is that relation correct?

Reply:
I think so. It's very weird, isn't it?

Reply:
Ya, I thought it should be `has_one` instead.

Reply:
I always have trouble remembering whether it's `belongs_to` or `has_one`. The diagrams at https://guides.rubyonrails.org/v4.2/association_basics.html actually help me remember. The gist is that `belongs_to` is used on a class when they have a field holding association (in this case, Supporter has a `primary_address_id` field). `has_one` would be used when the class at the other end of the association has the field.